### PR TITLE
[NET-1151 NET-11228] security: Add request normalization and header match options to prevent L7 intentions bypass

### PR DESCRIPTION
**Pull request by zalimeni**
### Description

This PR brings in all previously reviewed changes from  the `zalimeni/feature/net-1151-l7-intentions-security-fixes` feature branch into `main` and release branches. All changes were previously approved as part of Enterprise reviews except for the changelog added in this PR.

Changes include:

* [mesh: add options for HTTP incoming request normalization](https://github.com/hashicorp/consul/pull/21816/commits/bc988a340e8a14582d1659dc2ede0b76a597ecd3)
* [mesh: enable inbound URL path normalization by default](https://github.com/hashicorp/consul/pull/21816/commits/3cf889f3fbd5b2e4ee4e50463225b0f52db1c78c)
* [mesh: add support for L7 header match contains and ignore_case](https://github.com/hashicorp/consul/pull/21816/commits/b3dd745a5d6c3bfcb69cdf3e0e7c2ce7d6ccd2da)
* [ui: support L7 header match contains and ignore_case](https://github.com/hashicorp/consul/pull/21816/commits/dc5f65710dc7102298e2f4e331037e2f547e3fc6)
* [test: add request normalization integration bats tests](https://github.com/hashicorp/consul/pull/21816/commits/54bc8bb313f2e024965ee75af0237e6754ae2079)
* [docs: update security and reference docs for L7 intentions bypass pre…](https://github.com/hashicorp/consul/pull/21816/commits/947d789f2746d51385950d4f718b03830aa7c25e)

I'll squash and rebase these commits prior to merge to make backports more manageable.

Once this PR is merged, I'll cut `api` across active release branches, which will allow for https://github.com/hashicorp/consul-k8s/pull/4385 to be updated and merged as well, completing the cross-repo changeset.

### Testing & Reproduction steps

See previous PRs for testing details. All unit and integration tests are expected to pass.

### PR Checklist

* [x] updated test coverage
* [x] external facing docs updated
* [x] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

**Comment by zalimeni** [comment]
Fixed whitespace causing issues w/ docs mesh config entry render, and incorrect copilot-ed defaults I missed in the last two entries.

File: .changelog/21816.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+```release-note:security
+mesh: Add `http.incoming.requestNormalization` to Mesh configuration entry to support inbound service traffic request normalization.
+```
+```release-note:security
+mesh: Add `contains` and `ignoreCase` to L7 Intentions HTTP header matching criteria to support configuration resilient to variable casing and multiple values.
+```
+```release-note:breaking-change
+mesh: Enable Envoy `HttpConnectionManager.normalize_path` by default on inbound traffic to mesh proxies.
```
Comment (dduzgun-security) [inline .changelog/21816.txt:8]: `CVE-2024-10005` has been reserved for this, we could add that it resolves it in the changelog.

File: .changelog/21816.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+```release-note:security
+mesh: Add `http.incoming.requestNormalization` to Mesh configuration entry to support inbound service traffic request normalization.
+```
+```release-note:security
+mesh: Add `contains` and `ignoreCase` to L7 Intentions HTTP header matching criteria to support configuration resilient to variable casing and multiple values.
```
Comment (dduzgun-security) [inline .changelog/21816.txt:5]: `CVE-2024-10006` has been reserved for this, we could add that it resolves it in the changelog.

**Review by dduzgun-security — APPROVED** [review]
Awesome work, LGTM! 🙌 Only 2 suggestions for the changelogs.

**Review by ndhanushkodi — APPROVED** [review]

File: .changelog/21816.txt (hunk 1 above)
Comment (zalimeni) [reply .changelog/21816.txt:8]: Absolutely, I should have noted I'd planned on that 👍🏻 doing

**Comment by zalimeni** [comment]
Rebased and squashed for merge

**Comment by zalimeni** [comment]
Fixed minor docs misalignment from docs PR -> api/agent structs

**Pull request merged**